lib/saint.js

```diff
@@ -79,15 +79,23 @@ p.getJobSegment = function(fileId, pageNum, callback){
 	});
 }
 
-p.importJob = function(jobDescription, jobData, callback){
-	var self = this,
-			populateData =[];
-	while(jobData.length){
-		populateData.push(jobData.splice(0,10000));
+p.importJob = function(jobDescription, jobData, callback, jobId){
+	var self = this;
+	var populateData = spliceJobData(jobData);
+	function spliceJobData(jobData) {
+		var splicedJobData = [];
+		while(jobData.length){
+			splicedJobData.push(jobData.splice(0,10000));
+		}
+		return splicedJobData;
 	}
-
 	self.logger("info", "Creating Job");
-	this.importCreateJob(jobDescription, function(err, data){
+	if (typeof jobId === 'undefined') {
+		this.importCreateJob(jobDescription, fnCallImportPopulateJob);
+	} else {
+		fnCallImportPopulateJob(null, jobId);
```

Comment:
It seem clear that this is a function. What benefit do you get from the fn at the begining?

Reply:
I can change this. I thought the fn added more clarity that I was passing a function.